qiskit/algorithms/time_evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -84,7 +84,9 @@ def lse_solver(self) -> Callable[[np.ndarray, np.ndarray], np.ndarray]:
     def lse_solver(self, lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] | None) -> None:
         """Sets an LSE solver. Uses a ``np.linalg.lstsq`` callable if ``None`` provided."""
         if lse_solver is None:
-            lse_solver = lambda a, b: np.linalg.lstsq(a, b, rcond=1e-2)[0]
+
+            def lse_solver(a, b):
+                return np.linalg.lstsq(a, b, rcond=1e-2)[0]
```

Comment:
I don't black's spacing rule here  either. What about the argument that diagnostics are improved because the function has a name?  In some cases, the readability was improved by this rule. In others maybe not. In any case, for this PR, I am pulling out anything that is objected to. The idea is to evaluate using ruff, not to litigate rules.

Reply:
Reverted here: f02371b5897593721cf2191041d631a3770f11ac  adcd907da65ebc465429ee8f9019d259d69df313 d590b72a056d03301361ad0115e9d8a71957c267